Make -Dquickly a bit faster

gsmet opened the pull request:
Instantiation the plugins can actually be costly when we have so many artifacts.
For now, I have only applied this trick where it makes a great difference and has a low cost of maintenance.
We could also do it for Surefire.

Per a discussion with @ppalaga.

ppalaga commented on build-parent/pom.xml:
```diff
@@ -525,6 +525,7 @@
                     <executions>
                         <execution>
                             <id>enforce</id>
+                            <phase>${maven-enforcer-plugin.phase}</phase>
```
This is a neat trick! I have not realized it can be set through a property.

ppalaga commented:
Yeah, Surefire, Failsafe, impsort, formatter and even quarkus-maven-plugin:build are other good candidates to be handled in the same way.

gsmet committed:
Make -Dquickly a bit faster

Instantiation the plugins can actually be costly when we have so many
artifacts.
For now, I have only applied this trick where it makes a great
difference and has a low cost of maintenance.
We could also do it for Surefire.

Per a discussion with @ppalaga.
